frog: init at v0.13.7

roberth opened the pull request:
###### Motivation for this change

frog is a natural language parser for dutch.
It depends on some other packages in the LanguageMachines organisation.
I have included an update script.

###### Things done

Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers.

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

roberth committed:
frog: init at v0.13.7

Mic92 commented on pkgs/top-level/all-packages.nix:
```diff
@@ -8432,6 +8432,8 @@ with pkgs;
   };
   libkrb5 = krb5Full.override { type = "lib"; };
 
+  languageMachines = import ../development/libraries/languagemachines/packages.nix { inherit callPackage; };
```
As it is not part of the main namespace, the packages will be not build by hydra. Are you ok with this?

roberth replied:
That's a very good point! I will make frog available at the top level. I don't think its dependencies are used a lot outside frog, so I think it doesn't make sense to put everything in the already big `all-packages.nix`.

Is there a way to also include my test without polluting `all-packages.nix`? (It's a simple `runCommand` derivation that makes sure frog works as expected without a mutable `frog.out`, lingering builder state, etc.)

Mic92 replied:
You can add it to `checkPhase`, so it will be run when the package is build.

FRidh replied:
> As it is not part of the main namespace, the packages will be not build by hydra

Its possible to use `recurseIntoAttrs`

roberth commented:
- Added `recurseIntoAttrs` to also build `test` and any other languagemachines projects that may be added in the future
 - Rebased onto master

Thanks @Mic92 and @FRidh for the review and help!

Mic92 commented:
There seems to be a derivation failing: 

```
$ nix-shell --keep-going -p languageMachines.test
bash: /nix/store/6jaydhdipi87z9i4p8czq46nj2ga04ni-frog-test: line 1: syntax error near unexpected token `('
```

roberth commented:
I would expect Travis CI to catch that. I will look into it.

roberth commented:
Alright, what happens is that nix-shell wants to run the output of the test. I have no idea why that happens, because for example

    nix-shell --keep-going -p languageMachines.frog

does not run the executable. I did't expect `nix-shell -p` to try to run text files as scripts. (Did you?)

The following works and seems sufficient:

    nix-build -A languageMachines.frog '<nixpkgs>'

I think the nix-shell behaviour might be a bug.

Mic92 commented:
I guess it is ok then.
